types/underscore/underscore-tests.ts

```diff
@@ -801,6 +815,28 @@ declare const extractChainTypes: ChainTypeExtractor;
 
 // Types
 
+// TypeOfCollection
+declare const listItem: _.TypeOfCollection<_.List<StringRecord>>;
+listItem; // $ExpectType StringRecord
+
+declare const arrayItem: _.TypeOfCollection<StringRecord[]>;
+arrayItem; // $ExpectType StringRecord
+
+declare const augmentedListItem: _.TypeOfCollection<AugmentedList>;
+augmentedListItem; // $ExpectType StringRecord
+
+declare const augmentedListLiteralItem: _.TypeOfCollection<AugmentedListLiteral>;
+augmentedListLiteralItem; // $ExpectType StringRecord
```

Comment:
This is the check that fails without the changes in this PR (the type of `augmentedListLiteralItem` ends up being `StringRecord | number | string`). In theory the check for similarly-structured `augmentedListItem` should also fail, but for some reason it does not.

Reply:
Since my description seems to imply that this is an issue introduced by TS 4.0 specifically, I should note that while I found that the JQuery type interface started being interpreted in an undesirable way when I upgraded from TS 3.6 to TS 4.0, this issue affected type literals in versions lower than that (probably all the way down to TS 2.8).

Reply:
Thanks!

Reply:
As a point of interest here, I had hoped to be able to keep these to a single line by putting the `$ExpectType`s on the same lines as the `declare const ...`, but unfortunately TS linting didn't enjoy that approach.